OpenRA.Mods.Common/HitShapes/Rectangle.cs

```diff
@@ -0,0 +1,81 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Drawing;
+using System.Linq;
+using OpenRA.Graphics;
+
+namespace OpenRA.Mods.Common.HitShapes
+{
+	public class RectangleShape : IHitShape
+	{
+		public WDist OuterRadius { get; private set; }
+
+		[FieldLoader.Require]
+		public readonly int2 TopLeft;
+
+		[FieldLoader.Require]
+		public readonly int2 BottomRight;
+
+		int2 size;
+		int2 center;
+
+		WVec[] combatOverlayVerts;
+
+		public RectangleShape() { }
+
+		public RectangleShape(int2 tl, int2 br)
+		{
+			TopLeft = tl;
+			BottomRight = br;
+		}
+
+		public void Initialize()
+		{
+			if (TopLeft.X >= BottomRight.X || TopLeft.Y >= BottomRight.Y)
+				throw new YamlException("TopLeft and BottomRight points are invalid.");
+
+			size = (BottomRight - TopLeft) / 2;
```

Comment:
nit: This is a quadrant size rather than the actual dimensions. Perhaps `quadrantSize`.